### PR TITLE
feat: Add Certificate Template using Wagtail CMS

### PR DESCRIPTION
**Pull request by asadiqbal08**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
- [ ] Migrations
  - [ ] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
fixes: #691 

#### What's this PR do?
The PR enables the wagtail part to setup certificates along with signatories.

#### How should this be manually tested?
- Checkout the branch.
- Run the migrations
- Create a super user and visit `CMS`.
- In Pages > HomePages > ADD CHILD PAGE to ensure certificates and signatories pages in wagtail.
- In Courses select course and add certificate, at bottom click on preview to get the preview of the certificate

#### Screenshots

screencapture-localhost-8013-cms-pages-13-edit-2022-07-25-10_59_31
screencapture-localhost-8013-cms-pages-add-cms-signatorypage-10-2022-07-25-10_58_59
screencapture-localhost-8013-cms-pages-3-2022-07-25-10_58_08

#### Demo Certificate Template Screenshot
screencapture-localhost-8013-cms-pages-44-demo-certificate-2022-07-29-04_42_43_PM

### CODE REVIEW

**Comment by asadiqbal08** [comment]
The PR is in progress, @aliraza-abbasi will jump over this and I will help him out to understand the missinng areas. 

cc: @pdpinch

**Comment by pdpinch** [comment]
Can we get a screenshot of the learner's view of their own certificate, and a screenshot of an anonymous view of the certificate? If they are the same, we will need to work on #692 as a follow-up.

File: cms/templatetags/image_version_url.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+register = template.Library()
+
+
+@register.simple_tag()
+def image_version_url(
+    image, filter_spec, full_url=False, viewname="wagtailimages_serve"
+):
+    """
+    Generates an image URL using Wagtail's library and appends a version to the path to enable effective caching
```
Comment (arslanashraf7) [inline cms/templatetags/image_version_url.py:16]: I think we created this for caching wagtail images + image optimizations. And for that, we also used wagtail caching backend settings that can be seen [here in xPRO](https://github.com/mitodl/mitxpro/blob/master/mitxpro/settings.py#L912-L927). We have seen some issues with image caching though. So, I'm not sure if we want to introduce caching here or just use `image_url` from `wagtailimages_tags` in HTML.

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
I just left a couple of comments, I was able to functionality test this too and was able to create a certificate template in CMS just like xPRO.

However, In general, we have used duplicate assets in the PR. e.g.
1. Notice images `certificate-logo.png` and `certificate.png` (Duplicates)
2. Notice, `static/images/certificates/certificate-mit-logo.png` and `static/images/certificates/transparent-mit-xonline-logo.png` (I think already have a transparent MITx Online logo [here](https://github.com/mitodl/mitxonline/blob/main/static/images/mitx-online-logo.png)) that we can use.

File: cms/models.py [hunk 2]
```diff
@@ -59,6 +75,318 @@
...
+        blank=False,
+        help_text="Specify the course/program name.",
+    )
+
+    CEUs = models.CharField(
+        max_length=250,
+        null=True,
+        blank=True,
+        help_text="Optional text field for CEU (continuing education unit).",
+    )
```
[the remark this answers is not in the document]
Comment (pdpinch) [reply cms/models.py:257]: We don't need it, but it's OK to leave it in, for the sake on consistency across applications.

File: cms/templatetags/image_version_url.py (hunk 1 above)
Comment (pdpinch) [reply cms/templatetags/image_version_url.py:16]: I don't want to introduce image caching as a side-effect of this PR. And there's nothing about certificates by themselves that recommends caching. 

Can we leave it out without to much trouble?

File: cms/models.py [hunk 3]
```diff
@@ -59,6 +75,318 @@
...
+
+    @route(r"^$")
+    def index_route(self, request, *args, **kwargs):
+        """
+        The index page is not meant to be served/viewed directly
+        """
+        raise Http404()
+
+
+class CourseProgramChildPage(Page):
```
[the remark this answers is not in the document]
Comment (aliraza-abbasi) [reply cms/models.py:174]: Yes, It is used and it'll be used for program certification as well

**Comment by aliraza-abbasi** [comment]
> I just left a couple of comments, I was able to functionality test this too and was able to create a certificate template in CMS just like xPRO.
> 
> However, In general, we have used duplicate assets in the PR. e.g.
> 
>     1. Notice images `certificate-logo.png` and `certificate.png` (Duplicates)
> 
>     2. Notice, `static/images/certificates/certificate-mit-logo.png` and `static/images/certificates/transparent-mit-xonline-logo.png` (I think already have a transparent MITx Online logo [here](https://github.com/mitodl/mitxonline/blob/main/static/images/mitx-online-logo.png)) that we can use.

Removed duplicates

**Comment by arslanashraf7** [comment]
@aliraza-abbasi The PR has some conflicts, Can you rebase ?

File: cms/templates/certificate_page.html [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    <meta name="twitter:card" content="summary_large_image" />
+    <meta name="twitter:site" content="@MITxonedX" />
+    <meta name="twitter:image" content="{{ share_image_url }}" />
+    <meta property="og:title" content="{{ site_name }} | Certificate for: {{ page.product_name }}" />
+    <meta property="og:description" content="Certificate for {{ page.product_name }} awarded by {{ site_name }}." />
+    <meta property="og:url" content="{{ request.build_absolute_uri }}" />
+    <meta property="og:image:url" content="{{ share_image_url }}" />
+    <meta property="og:image:width" content="{{ share_image_width }}" />
+    <meta property="og:image:height" content="{{ share_image_height }}" />
+    <meta property="og:image:alt" content="A certificate from MIT xOnline" />
```
Comment (arslanashraf7) [inline cms/templates/certificate_page.html:19]:
```suggestion
    <meta property="og:image:alt" content="A certificate from MITx Online" />
```

File: main/urls.py [hunk 5]
```diff
@@ -25,6 +25,8 @@
 
 from main.views import cms_signin_redirect_to_site_signin, index, refine
 
+WAGTAIL_IMG_CACHE_AGE = 31_536_000  # 1 year
```
Comment (arslanashraf7) [inline main/urls.py:28]: We might not need this.

**Review by arslanashraf7 — CHANGES_REQUESTED** [review]
A couple of quick changes + below:

1.  We have added `share-image.png` (I think we are not using it anywhere also it says xPRO in the image), If we are not using it we should remove it from the PR.
2. We also have a `sign.png` which isn't used anywhere I suppose.
3. Can you rebase the PR also, there are recent updates in main branch that might have conflicts with your migrations in courses.

**Review by arslanashraf7 — APPROVED** [review]
LGTM 👍

File: cms/models.py [hunk 6]
```diff
@@ -1,20 +1,21 @@
 """CMS model definitions"""
 import logging
 import re
-import datetime
+from datetime import datetime, timedelta
```
Comment (asadali145) [inline cms/models.py:4]: This change is causing issues. 
Screenshot 2022-08-17 at 7 36 23 PM

**Pull request merged**